src/util/RequestUtil.php

```diff
@@ -54,8 +54,10 @@ public static function checkNameParam() {
             exit();
         }
 
-        if (!ctype_alnum($name)) {
-            echo new Error(400, "Invalid name. Name must be alphanumeric.");
+        // regex as per https://github.com/SpigotMC/XenforoResourceManagerAPI/issues/41#issuecomment-861888191
+        // slightly modified to accommodate legacy usernames
+        if (!preg_match("/^[A-Za-z0-9_\.\- ]{3,}$/", $name)) {
+            echo new Error(400, "Invalid name. Name must consist of letters, numbers, and/or a limited set of special characters (_, -, ., and/or one or more).");
```

Comment:
The regex above only matches if there are 3 or more characters.
A hint in the error message for the user would be good.

Reply:
Good idea! Thank you!

Updating to "Invalid name. Name must be at least three characters in length and consist of letters, numbers, and/or a limited set of special characters (_, -, ., and/or one or more spaces)."